ldk/javascript/readme.md

```diff
@@ -28,6 +28,43 @@ const merged = merge.merge(ldkConfig.default, {
 module.exports = merged;
 ```
 
+### Loop Permissions
+In order to ensure your Loop is executing in a secure manner, you must declare which network URLs, file system paths, and aptitudes your Loop will use.
+
+Permissions are declared inside of the Loop `package.json` within a `permissions` json object.
```

Comment:
It's very possible that we might find additional LDK metadata we'd want to put in here. I'm wondering if instead of calling this property `permissions`, that we call it `ldk` and nest the `permissions` object under that.

Reply:
sold

Reply:
@iangallOlive I'd like your input here as well.

Reply:
Yep I think this make sense (top-level key is `ldk`, with a child `permissions` key)

Reply:
What will that look like as JSON in once it's all in the header? We're looking right now for

```
{
   "ldkVersion": "semver", // Soon to be "oliveHelpsContractVersion"
}
```

I think additional top level keys are ideal (from our perspective) here. That is, I wouldn't want `ldk` to also appear in this final output. Just the _value_ of that property. So something like

```
{
  "oliveHelpsContractVersion": "semver",
  "permissions": {
    "urlPermissions": "...",
    "filesystemPermissions": "...", // should it be fileSystemPermisions???
    "aptitudePermissions": "..."
  }
}
```

Reply:
To clarify, there's three reasons for that. 1) `ldk` would make the nesting kinda crazy. You'd need `loop.customData.webpackMetadata.ldk.permissions`. 2) It makes validating it a little trickier. Instead of just "does this object have a, b, and c in it?" we have to ask "does this object have a and b and does b also have c and d?" 3) As you can see we're already grouping all of this as `webpackMetadata` because from our perspective, it's just loop-specific custom data, and that fact that it comes from or belongs to the ldk specifically seems like something the loop library doesn't need to know or deal with. From our point of view, it's _all_ ldk in a way.

Reply:
I think the `ldk` key was simply to make it obvious in the `package.json` that everything inside of it is for "Loop stuff". So I'd agree, and I think it was the intent, that only its **contents** would be used.

Reply:
We can use `ldk` at the top level, and send metadata withough wrapping it within `ldk` if that works for everyone. I do like the idea of separation within `package.json`.